### ISSUE TITLE
Generator

### ISSUE THREAD

**Issue by juliocbcotta** [flutterconsortium/dart_sealed_unions#11]
Hey, how about to add a code generator to avoid the huge amount of boiler plate ?

**Comment by nodinosaur** [comment]
It is a good idea, unfortunately I have a lot of family related priorities at the moment and thus my time is extremely limited.
But I am happy to review any pull requests which are welcome.

**Comment by juliocbcotta** [comment]
I will try a standalone package for now, if it works well we can merge it latter.
Thanks.

**Comment by juliocbcotta** [comment]
I gave it a try and i got something working..I think..
from this [definition](https://github.com/BugsBunnyBR/sealed_unions_generator.dart/blob/master/example/lib/weather_model.dart) I got [this](https://github.com/BugsBunnyBR/sealed_unions_generator.dart/blob/master/example/lib/weather_model.g.dart) generated code.
I know my approach has some downsides and probably some issues..
1) There is no actual implementation for that model.. I only used it as a template to generate the code.
2) I am new to sealed unions and I failed to understand how I could map that definition to a Nullet... I don't know if there is a use-case for that when trying to map a State...

Any way, I will publish this on pub.dev soon, if by chance anyone can take a look at the library it would be great to have some feedback.
